packages/aws-cdk/lib/api/evaluate-cloudformation-template.ts

```diff
@@ -33,7 +34,57 @@ export class LazyListStackResources implements ListStackResources {
   }
 }
 
-export class CfnEvaluationException extends Error {}
+export interface LookupExport {
+  lookupExport(name: string): Promise<AWS.CloudFormation.Export | undefined>;
+}
+
+export class LookupExportError extends Error { }
+
+export class LazyLookupExport implements LookupExport {
+  private exports: { [name: string]: AWS.CloudFormation.Export } = {}
+
+  constructor(private readonly sdk: ISDK) { }
+
+  async lookupExport(name: string): Promise<AWS.CloudFormation.Export | undefined> {
+    if (this.exports[name]) {
+      return this.exports[name];
+    }
+
+    for await (const exports of this.listExports()) {
+      for (const e of exports) {
+        if (!e.Name) {
+          throw new LookupExportError(`Unable to handle CloudFormation Export without Name! ${JSON.stringify(e)}`);
+        }
+        this.exports[e.Name] = e;
+
+        if (e.Name === name) {
+          return e;
+        }
+      }
+    }
+
+    return undefined; // export not found
+  }
+
+  private async * listExports() {
+    let nextToken: string | undefined = undefined;
+    let response: PromiseResult<AWS.CloudFormation.ListExportsOutput, AWS.AWSError>;
```

Comment:
Moved to a `const` within the loop, but unable to remove the type def 👇 

```
'response' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer. ts(7022)
```